bessctl/module.py

```diff
@@ -22,10 +22,21 @@ def __init__(self, **kwargs):
         else:
             name = None
 
-        ret = self.bess.create_module(self.__class__.__name__, name,
+        if(not _do_not_create):
+            #create an object
+            ret = self.bess.create_module(self.__class__.__name__, name,
                                       self.choose_arg(None, kwargs))
-
-        self.name = ret.name
+            self.name = ret.name
+        else:
+            #bind to a pre-existing object, check if it's real
+            assert self.name != None
+            all_modules = self.bess.list_modules().modules
```

Comment:
`list_modules()` enumerates all modules, thus potentially expensive if there are a lot. Instead you can use (untested):
```
info = self.bess.get_module_info(name)
check if self.__class__.__name__ agrees with info.mclass
```